Allow client_id to be configured on vault_identity_oidc_role resources

ianferguson opened the pull request:
<!--- Please keep this note for the community --->

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### Allow client_id to be configured on vault_identity_oidc_role resources

[Vault v1.4.0](https://github.com/hashicorp/vault/blob/master/CHANGELOG.md#140-april-7th-2020) added support for configuring the `client_id` of OIDC Identity Roles (via https://github.com/hashicorp/vault/pull/8165)

This pull request adds support to the `vault_identity_oidc_role` resource for configuring the `client_id` parameter now supported by Vault v1.4.0+

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-vault/blob/master/CHANGELOG.md): 

```release-note
* resource/vault_identity_oidc_role: `client_id` parameter can optionally be configured ([#815](https://github.com/terraform-providers/terraform-provider-vault/pull/815)).
```

#### Output from acceptance testing:

- existing integration test with no client_id set to help ensure backwards compatibility was maintained
- integration test with a configured client_id was added
- added client_id update to the existing integration test of updating the resource

```
> make testacc TESTARGS='-run TestAccIdentityOidcRole.*'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run TestAccIdentityOidcRole.* -timeout 120m
?       github.com/terraform-providers/terraform-provider-vault [no test files]
?       github.com/terraform-providers/terraform-provider-vault/cmd/coverage    [no test files]
?       github.com/terraform-providers/terraform-provider-vault/cmd/generate    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/codegen 0.025s [no tests to run]
?       github.com/terraform-providers/terraform-provider-vault/generated       [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/generated/datasources/transform/decode  0.037s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/generated/datasources/transform/encode  0.037s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/generated/resources/transform/alphabet  0.037s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/generated/resources/transform/role      0.106s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/generated/resources/transform/template  0.036s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/generated/resources/transform/transformation    0.031s [no tests to run]
?       github.com/terraform-providers/terraform-provider-vault/schema  [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/util    0.037s [no tests to run]
=== RUN   TestAccIdentityOidcRole
--- PASS: TestAccIdentityOidcRole (0.60s)
=== RUN   TestAccIdentityOidcRoleWithClientId
--- PASS: TestAccIdentityOidcRoleWithClientId (0.40s)
=== RUN   TestAccIdentityOidcRoleUpdate
--- PASS: TestAccIdentityOidcRoleUpdate (0.46s)
PASS
ok      github.com/terraform-providers/terraform-provider-vault/vault   1.490s
```

ianferguson commented:
I know pull request reviews on this repository are done periodically, but was wondering if there was an estimate when the next round of reviews might be, and if that would include this PR?

ianferguson commented:
@Valarissa would it be possible for this PR to be reviewed in the coming days or weeks?

ianferguson commented:
@jasonodonnell would it be possible for this PR to be reviewed and hopefully included in a future release?

jasonodonnell commented:
Hi @ianferguson! Yes, I will get this reviewed sometime next week.

jasonodonnell commented on vault/resource_identity_oidc_role_test.go:
```diff
@@ -209,7 +242,22 @@ resource "vault_identity_oidc_role" "role" {
+resource "vault_identity_oidc_role" "role" {
+	name = "%s"
+	key = vault_identity_oidc_key.key.name
+    client_id = "%s"
```
```suggestion
	client_id = "%s"
```

jasonodonnell commented on vault/resource_identity_oidc_role_test.go:
```diff
@@ -219,10 +267,11 @@ resource "vault_identity_oidc_key" "key" {
 resource "vault_identity_oidc_role" "role" {
 	name = "%s"
 	key = vault_identity_oidc_key.key.name
+    client_id = "%s"
```
```suggestion
	client_id = "%s"
```

jasonodonnell reviewed (approved):
Minor formatting but LGTM!

ianferguson commented:
@jasonodonnell thanks for the review! I've addressed the feedback, updated the changelog note to be in the now current release that is queued up and rebased on to the current `master` branch in df130a79,

Let me know if there is anything else I can do to help get this merged!

jasonodonnell commented:
@ianferguson Can you remove the CHANGELOG edit? We'll update that once we merge this (you can see there's a conflict now).

ianferguson commented:
@jasonodonnell done in 71ad7f38f7d09536c973009fb1f7e44c7827e3f3

jasonodonnell commented:
Thanks @ianferguson!
